### PR TITLE
Add command `truffle config list`

### PR DESCRIPTION
**Pull request by MrGrizz**
This PR add command 'truffle config list' mentioned in #4379

### CODE REVIEW

**Comment by gnidan** [comment]
Love this attention to `truffle config`! Thank you!

**Review by cds-amal — COMMENTED** [review]
Thanks for this @MrGrizz! I have one issue with the PR around git practice. Can you rebase the last commit against upstream develop instead of merging?

**Comment by MrGrizz** [comment]
@cds-amal Done. Sorry for that mistake.

File: packages/core/lib/commands/config.js [hunk 1]
```diff
@@ -158,6 +166,11 @@ const parse = function (args) {
 
       break;
     }
+    case "list": {
+      set = false;
```
Comment (eggplantzzz) [inline packages/core/lib/commands/config.js:170]: You shouldn't need to set the value for `set` as it is already `false`. Actually I have no idea why this happens on line 120 either.

File: packages/core/lib/commands/config.js [hunk 2]
```diff
@@ -21,6 +21,10 @@ const command = {
       {
         option: "set",
         description: "Set a Truffle config option value."
+      },
+      {
+        option: "list",
+        description: "List a Truffle config option values.",
```
Comment (eggplantzzz) [inline packages/core/lib/commands/config.js:27]: So I think this should be "List all Truffle config values." That's what this feature is meant to do, right? List all the stored Truffle config values?

File: packages/core/lib/commands/config.js [hunk 3]
```diff
@@ -73,6 +77,9 @@ const command = {
       }
 
       return;
+    } else if (command.list) {
+      options.logger.log("; Truffle configs");
```
Comment (eggplantzzz) [inline packages/core/lib/commands/config.js:81]: And this should probably read "Truffle config values"

File: packages/core/lib/commands/config.js (hunk 1 above)
Comment (MrGrizz) [reply packages/core/lib/commands/config.js]: Done.

File: packages/core/lib/commands/config.js (hunk 2 above)
Comment (MrGrizz) [reply packages/core/lib/commands/config.js]: You're right. Done.

**Comment by eggplantzzz** [comment]
I have one or two more nitpicks and then this should be good to go as far as I'm concerned. Thanks!

File: packages/core/lib/commands/config.js [hunk 4]
```diff
@@ -73,6 +77,9 @@ const command = {
       }
 
       return;
+    } else if (command.list) {
+      options.logger.log("; Truffle config values");
```
Comment (eggplantzzz) [inline packages/core/lib/commands/config.js:81]: Could you remove the `; ` from the beginning of this string? I changed it locally to look like this at the command line.

Screen Shot 2021-12-02 at 10 36 53 AM

File: packages/core/lib/commands/config.js [hunk 5]
```diff
@@ -21,6 +21,10 @@ const command = {
       {
         option: "set",
         description: "Set a Truffle config option value."
+      },
+      {
+        option: "list",
+        description: "List all Truffle config values",
```
Comment (eggplantzzz) [inline packages/core/lib/commands/config.js:27]: Can you add a `.` at the end of this sentence. That seems to be convention for all descriptions.

File: packages/core/lib/commands/config.js (hunk 4 above)
Comment (MrGrizz) [reply packages/core/lib/commands/config.js]: Removed. I just add `;` because I followed `npm config list` :D

**Review by eggplantzzz — APPROVED** [review]
Thanks @MrGrizz, I like it!

**Review by cds-amal — APPROVED** [review]
LGTM! Thanks @MrGrizz

**Pull request merged**